hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/stats/ExpressionEvaluator.java

```diff
@@ -282,6 +286,9 @@ public static LessThan getInstance() {
 
     @Override
     public boolean eval() {
+      if (this.val == null) {
+        return false;
+      }
```

Comment:
Can we pre-process these NULLs before entering specific evaluators?

Reply:
Sure.
I introduce a new evaluator to handle this.